compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -4038,7 +4036,14 @@ pub(crate) enum ForbiddenLetReason {
     NotSupportedParentheses(#[primary_span] Span),
 }
 
-/// Visitor to check for invalid/unstable use of `ExprKind::Let` that can't
+/// Whether let chains are allowed on all editions, or it's edition dependent.
+/// In case of edition dependence, specify the currently present edition.
+pub enum LetChainsPolicy {
+    AlwaysAllowed,
+    Edition(Edition),
+}
```

Comment:
```suggestion
/// Whether let chains are allowed on all editions, or it's edition dependent.
/// In case of edition dependence, specify the currently present edition.
pub enum LetChainsContext {
    AlwaysAllowed,
    EditionDependent { current_edition: Edition },
}
```

When you write it as `LetChainsPolicy` it feels like the edition that you are passing in is some sort of condition that must be passed, as compared to the current edition. I believe this change would help clarify things better.

Reply:
changed